kore/test/Test/Kore/Step/Simplification/TermLike.hs

```diff
@@ -41,22 +39,3 @@ simplifyEvaluated original =
             -- Throw an error if any predicate would be simplified.
             ConditionSimplifier $ const undefined
         }
-
-newtype TestSimplifier a = TestSimplifier { getTestSimplifier :: Simplifier a }
-    deriving (Functor, Applicative, Monad)
-    deriving (MonadLog, MonadSMT)
-
-instance MonadSimplify TestSimplifier where
-    askMetadataTools = TestSimplifier askMetadataTools
-    askSimplifierAxioms = TestSimplifier askSimplifierAxioms
-    localSimplifierAxioms f =
-        TestSimplifier . localSimplifierAxioms f . getTestSimplifier
-    askMemo = TestSimplifier (Memo.liftSelf TestSimplifier <$> askMemo)
-    askInjSimplifier = TestSimplifier askInjSimplifier
-    askOverloadSimplifier = TestSimplifier askOverloadSimplifier
-    simplifyCondition sideCondition condition =
-        Logic.mapLogicT TestSimplifier
-        (simplifyCondition sideCondition condition)
-
-    -- Throw an error if any term would be simplified.
-    simplifyTermLike = undefined
```

Comment:
The compiler was complaining that `TestSimplifier` isn't used anywhere. @ttuegel you added this in the profiling PR, right?